src/driver/franka_plan_runner.cc

```diff
@@ -441,9 +441,7 @@ bool FrankaPlanRunner::RecoverFromControlException() {
                         utils::RobotModeToString(current_mode))};
         dexai::log()->error("RecoverFromControlException: {}", err_msg);
         comm_interface_->SetDriverIsRunning(false, err_msg);
-        // if error recovery fails because we are in the wrong mode, need user
-        // to intervene, so wait
-        std::this_thread::sleep_for(std::chrono::milliseconds(200));
+        return false;
```

Comment:
why fail instead of wait now? since we send the right message to dig? will we no longer resume recovery once we're back in the right mode?

Reply:
The logic is duplicated elsewhere 
```c++
          while (!RecoverFromControlException()) {  // plan_ is released/reset
            // keep trying to recover, expect this will require manual
            // intervention - this only fails when robot is user stopped or
            // locked
            // Set robot state here so we're still publishing an accurate state
            const auto robot_state {robot_->readOnce()};
            const auto canonical_robot_state {
                utils::ConvertToCannonical(robot_state, joint_pos_offset_)};
            comm_interface_->SetRobotData(canonical_robot_state,
                                          next_conf_plan_, franka_time_,
                                          plan_utime_, plan_start_utime_);
            std::this_thread::sleep_for(std::chrono::milliseconds(200));
          }
```